src/software/ai/navigator/path_planner/enlsvg_path_planner.h

```diff
@@ -52,6 +52,18 @@ class EnlsvgPathPlanner : public PathPlanner
                                  const Rectangle &navigable_area,
                                  const std::vector<ObstaclePtr> &) override;
 
+    /**
+     * Returns a path that is an optimized path between start and end. Has no checking on
+     * whether the start and end are valid and within field boundaries.
+     *
+     * @param start  start point
+     * @param end    end point
+     *
+     * @return   a vector of points that is the optimal path avoiding obstacles
+     *           if no valid path, then return an empty vector
+     */
+    std::optional<Path> findPath(const Point &start, const Point &end);
```

Comment:
with this function, can we make this class not a subclass of PathPlanner

Reply:
yes